### ISSUE TITLE
Using tint-color on images not working properly

### ISSUE THREAD

**Issue by sgrgrsn** [NativeScript/NativeScript#4778]
I am experiencing some issues when using the tint-color CSS property on Image widgets. For explaining my issue I have created a new demo project with the angular template `tns create TintColor --ng`.

On iOS the problem is that the images are not tinted before you have navigated to another view, and navigate back (see the example below). On Android the images are not tinted at all.

I have tried to use the CSS-selector `tint-color: #FF0000;` from a CSS-file and also with an inline style attribute. I have also tried to do it with the `tintColor="#FF0000"` attribute directly on the element.

### Code examples
Here are some of the ways I have tried to do the tinting. Check the demo project for complete source code.

**With CSS**
```html
<Image src="res://ic_access_time_black_24dp"></Image>
```
 
```css
Image {
     tint-color: #FF0000;
}
```

**With inline CSS**
```html
<Image src="res://ic_access_time_black_24dp" style="tint-color: #FF0000;"></Image>
```

**With the tintColor attribute**
```html
<Image src="res://ic_access_time_black_24dp" tintColor="#FF0000"></Image>
```

The problem occurs on both emulators and devices.

### Version numbers:
- CLI: `3.1.3`
- Cross-platform modules: `3.1.1`
- Runtime(s): `iOS: 3.1.0` and `Android: 3.1.1`
- Plugin(s): `nativescript-angular: 4.2.0` 

## Examples and source code
I have attached the source code for the demo project and a GIF showing how it looks on iOS.

### Demo project
[TintColorIssueDemo.zip](https://github.com/NativeScript/NativeScript/files/1263282/TintColorIssueDemo.zip)

### GIF from iOS - no images in the list items are tinted before you navigate back

![example1](https://user-images.githubusercontent.com/6108577/29868517-7de58602-8d7f-11e7-8dfa-2fa87551075e.gif)

**Comment by tsonevn** [comment]
Hi @sgrgrsn,
Thank you for reporting this issue. 
I tested your case on my side and indeed there is an issue with applying the tint-color to the Image. 
This problem also is reproducible with a non-Angular project. I am attaching sample project, which could be used for debugging.

The project is tested also with @next modules.
Regarding that, for further info, you could keep track on the issue.
[Archive.zip](https://github.com/NativeScript/NativeScript/files/1263532/Archive.zip)

**Comment by ddfreiling** [comment]
I've found the reason for the issue on iOS:
https://github.com/NativeScript/NativeScript/blob/02dba02e8bd35a270d6e2a8e4db0a18ec89c5615/tns-core-modules/ui/image/image.ios.ts#L27-L46

On line 31 the image is reset to a non-tinted version, but it does so whenever setTintColor is called a second time. It seems to me this reset should only happen when `value` is a non-color.

As for Android, where tintColor just does not work, I found out that Nativescript is using its own ImageView subclass with a custom `draw()` method, which paints the DrawableBitmap using a BitmapShader. In all likelihood someone forgot to apply the colorFilter to the shader in this custom draw method. The solution should probably be done in this part of tns-core-modules-widgets:
https://github.com/NativeScript/tns-core-modules-widgets/blob/8f4f938d248e2bd253523bf31b641998078d9f1c/android/widgets/src/main/java/org/nativescript/widgets/ImageView.java#L308-L312

I'll make a PR for this once I figure out how to fix tintColor on Android.
@PanayotCankov Maybe you have an idea, seems you wrote parts of the android ImageView code.
